imports/server/announcements.ts

```diff
@@ -4,6 +4,40 @@ import { Roles } from 'meteor/nicolaslopezj:roles';
 import Ansible from '../ansible';
 import Announcements from '../lib/models/announcements';
 import PendingAnnouncements from '../lib/models/pending_announcements';
+import Profiles from '../lib/models/profiles';
+
+Meteor.publish('announcements.pending', function () {
+  if (!this.userId) {
+    throw new Meteor.Error(401, 'Unauthenticated');
+  }
+
+  const pas = PendingAnnouncements.find({ user: this.userId });
+  const announcementIds = [...new Set(pas.map(pa => pa.announcement))];
+  const announcements = Announcements.find({ _id: { $in: announcementIds } });
+
+  const creatorIds = [...new Set(announcements.map(a => a.createdBy))];
+  const creators = Profiles.find({ _id: { $in: creatorIds } }, { fields: { displayName: 1 } });
+
+  return [pas, announcements, creators];
+});
+Meteor.publish('announcements.all', function (huntId: string) {
+  check(huntId, String);
+  const u = Meteor.users.findOne(this.userId);
+  if (!u) {
+    throw new Meteor.Error(401, 'Unauthenticated');
+  }
+
+  if (!u.hunts.includes(huntId)) {
+    throw new Meteor.Error(403, 'Not a member of that hunt');
+  }
+
+  const announcements = Announcements.find({ hunt: huntId });
+
+  const creatorIds = [...new Set(announcements.map(a => a.createdBy))];
+  const creators = Profiles.find({ _id: { $in: creatorIds } }, { fields: { displayName: 1 } });
+
+  return [announcements, creators];
```

Comment:
I believe this is not *quite* going to do what you want, since publishes are not rerun on result changes.

Namely, what will happen here is that if a new `Announcement` is created by someone who did not wind up in `creatorIds` when the subscription was created, that `Announcement` will get published to the client, but the `Profile` for the creator will not, which will wind up with missing data client-side when you go to render it.